src/controllers/userController.js

```diff
@@ -0,0 +1,163 @@
+require('dotenv').config();
+const admin = require('../config/firebase');
+const db = admin.firestore();
+const axios = require('axios');
+
+exports.signupUser = async (req, res) => {
+    // console.log(req.body)
+    const { name, email, password, IsAdmin } = req.body;
+
+    if (!name || !email || !password) {
+        return res.status(400).json({ error: 'All fields are required' });
+    }
+
+    try {
+        const userRecord = await admin.auth().createUser({
+            displayName: name,
+            email,
+            password
+        });
+        // console.log('User created:', userRecord);
+
+        if (IsAdmin) {
+            await admin.auth().setCustomUserClaims(userRecord.uid, { admin: true });
+            // console.log('Admin claim set for user:', userRecord.uid);
+        }
+
+        await db.collection('users').doc(userRecord.uid).set({
+            name,
+            email,
```

Comment:
Do we need to set email here if we already included it when creating the user?

Also, what's the different between name and displayName?

Reply:
Lines 26, 27, 28, and 29 are for creating a user during signup. and there is no difference between name and displayName, so I have made changes to remove displayName and keep only name to avoid confusion. Thank you